runtimes/alibi-detect/mlserver_alibi_detect/runtime.py

```diff
@@ -58,9 +65,11 @@ def __init__(self, settings: ModelSettings):
         super().__init__(settings)
 
     async def load(self) -> bool:
-        model_uri = await get_model_uri(self._settings)
+        self._model_uri = await get_model_uri(self._settings)
         try:
-            self._model = load_detector(model_uri)
+            self._model = load_detector(self._model_uri)
+            # Check whether an online drift detector (i.e. has a save_state method)
+            self._online = True if hasattr(self._model, 'save_state') else False
```

Comment:
The presence of a `save_state` method implies the detector is an online drift detector, meaning `detector.predict` takes one instance at a time **and** we should save state. Conversely, if the detector isn't an online drift detector, `state_save_freq=None` to prevent attempting to save state.

Note that checking `self._model.meta['online']` might be more elegant/readable. However, the meaning of this meta-field in Alibi Detect is currently confusing, since we set `online=True` for some outlier detectors such as `[Mahalanobis](https://docs.seldon.io/projects/alibi-detect/en/stable/od/methods/mahalanobis.html)`, despite this not having a `save_state` method and `predict` taking a batch of instances.

The above could need reconsidering in the future if we add `save_state` methods to new detectors, but we don't have immediate plans for this...